SNIPS/snip-86.md

```diff
@@ -0,0 +1,278 @@
+
+## Motivation
+
+SIWS aims to enhance user security by utilizing Starknet's account abstraction feature, allowing dynamic handling of signatures where any hash passing an account's `is_valid_signature` method is considered valid. This significantly improves security by enabling users to revoke signatures by changing account keys and supports future-proofing against advances in cryptographic methods. Additionally, by removing the character limit and employing structured JSON data for authentication requests, SIWS improves the clarity and detail of communication between users and services, making interactions more straightforward and secure. The flexibility offered by SIWS encourages developers to create innovative applications with diverse authentication needs, further enriching the Starknet ecosystem.
+
+## Specification
+
+### Sign-In Schema Structure
+
+Domain Object:
+
+- chainId: Identifies the Starknet network to confirm the correct context for authentication, aligning with protocol-level standards for network identification.
+- name: The application's name, clarifying for users which service is initiating a sign-in request, thus ensuring transparency.
+- version: Indicates the version of the application, ensuring that the authentication interaction adheres to compatible protocols.
+- revision: The Typed Data revision, '0' marks legacy Typed Data, while '1' indicates the active Typed Data standard.
+
+Message Object:
+
+- address: Specifies the wallet address responsible for verifying the user’s signature.
+- nonce: A one-time, unique number that makes each authentication request distinct to prevent replay attacks.
+- issuedAt: Timestamp indicating when the authentication request was generated, helping to establish its temporal validity.
+- statement: A user-readable message that the signer is agreeing to by providing their signature. This increases transparency and informs users about the purpose of their authentication.
+- uri: The identifier of the resource or the specific service that the authentication request is associated with, helping to contextualize the sign-in attempt.
+- version: The protocol version used in the message, ensuring compatibility between the user’s client and the dApp.
+- expirationTime (optional): The timestamp until which the authentication request remains valid, securing the process by limiting the time frame for a valid sign-in.
+- notBefore (optional): The earliest timestamp at which the authentication request can be considered valid, preventing its use before a certain time.
+
+### Sign-in Schema
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "domain": {
+      "type": "object",
+      "properties": {
+        "chainId": {
+          "type": "string",
+          "enum": ["SN_SEPOLIA", "SN_MAIN"],
```

Comment:
Why should we only limit it to two chains? There are a few appchains already (paradex for instance) and probably more soon; 
Moreover, you'd ideally want this standard to work on a local devnet too without using the devnet with testnet chainId imo

Reply:
Regarding this, the challenge would be to identify a standard format in which we can generalize over chainIDs such as EVM-chains where it starts with 1. I noticed that such a method does not exist yet with Starknet where it can be identified with an integer (might be wrong here), perhaps for a start we can allow these two first?